public/app/features/search/page/components/SearchView.tsx

```diff
@@ -84,6 +91,34 @@ export const SearchView = ({ showManage, folderDTO, queryText, hidePseudoFolders
     [searchSelection]
   );
 
+  const keyListener = useCallback((evt: KeyboardEvent) => {
```

Comment:
the hooks version of this feels pretty sketchy to me... but it seems to work, so maybe it is OK 🤷

Reply:
🤔 I am hesitant about this implementation, to fully support keyboard navigation we also should support navigating through the focusable elements in the table columns ( checkboxes, tags, etc)

Adding this  approach to our codebase makes it look more inconsistent :( (combination of tabs and arrow down and up but not really connected)

https://user-images.githubusercontent.com/239999/171038938-16adb465-7751-43f7-8c35-acd3d7094633.mp4

An ideal implementation with all a11y powers should be to do something like this (video [taken from react aria](https://react-spectrum.adobe.com/react-aria/useTable.html), a library that we use in some parts of Grafana)

https://user-images.githubusercontent.com/239999/171039604-23cb4590-a6a6-41b9-a7e9-0d1e447ae643.mp4

*ps,  I think we should take this video as an inspiration, I am hesitant to use this library again because it has proven to be quite complex to work with.*

Reply:
Testing react-aria I find it very confusing, especially if you want to just select the row you have to first tab through all columns. Pretty bad for search experience. This PR does not handle tab focus, this is PR just when using arrow keys when the input has focus, a common feature for almost all search boxes.